stackrox/stackrox
# chore(ui): Fix accessibility issues in clusters table and Next step

### Description

Because the number of changes, we will wait to merge until after 4.5 code freeze.

### Problems

> Interactive controls must not be nested

```html
<button aria-disabled="false" class="pf-v5-c-button pf-m-secondary" type="button" data-ouia-component-type="PF5/Button" data-ouia-safe="true" data-ouia-component-id="OUIA-Generated-Button-secondary-3">related node<a class="no-underline flex-shrink-0" data-testid="manageTokens" href="/main/clusters/init-bundles">Init bundles</a>
```

> Heading levels should only increase by one

```html
<h3 class="pf-v5-c-alert__title"><span class="pf-v5-screen-reader">Info alert:</span>Dynamic configurations are automatically applied</h3>
```

### Analysis

1. PatternFly `Button` contains `HashLink` element, therefore renders `button` that contains `a` element.

2. One step forward, one step backward.
    * Fixed heading levels `h3` of side panel which had `h2` element in #7325
    * Replaced side panel with cluster page which has `h1` element in #9390

### Solutions

1. Delete ManageTokens.tsx file.
    By the way, `HashLink` is obsolete after cluster init bundles moved from integrations to clusters route.
    * Edit ClustersTablePanel.tsx file to replace with same pattern as other links.
    * Edit CredentialExpirationWidget.tsx file to replace with `Link` element.

2. Fix again:
    * Replace `component="h3"` with `component="div"` in PatternFly `Alert` element.
    * Replace classic `CollapsibleCard` elements with PatternFly `Title` element that has `headingLevel="h2"` prop.
    * Add a heading level for **Download manifest bundle** that is parallel to **Download Helm values**.
    * Replace Tailwind layout with PatternFly `Flex` layout.
        Brad, thank you for example from compliance that gave me `flexWrap={{ default: 'nowrap' }}` counterpart for `direction={{ default: 'column', lg: 'row' }}` prop.
    * Replace classic `ToggleSwitch` with PatternFly `Switch` element.

### User-facing documentation


### Testing


#### Automated testing


#### How I validated my change

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js -68 = 4637042 - 4637110
        total -3463 = 11731113 - 11734567
    * `ls -al build/static/js/*.js | wc`
        files -1 = 174 - 175
3. `yarn start` in ui

### Manual testing

1. Visit /main/clusters

    * Before changes, see accessibility issue.
        ![clusters_with_issue](https://github.com/stackrox/stackrox/assets/11862657/1446b929-efce-44dc-ac15-4f541edfa4bd)

    * After changes, solve accessibility issue.
        ![clusters_without_issue](https://github.com/stackrox/stackrox/assets/11862657/4a3d5441-f69c-43a6-b785-9558e41ffb8c)

2. Click a row, and then click **Next**.

    * Before changes, see accessibility issue.
        ![clusters_id_Next_with_issue](https://github.com/stackrox/stackrox/assets/11862657/92134bbe-d54c-4c16-ba82-d1ba839d921d)

    * After changes, solve accessibility issue.
![clusters_id_Next_without_issues](https://github.com/stackrox/stackrox/assets/11862657/82bb953c-7c57-4593-9fbc-d59eaaad9869)

### Integration testing

* clusters/certificateExpiration.test.js
